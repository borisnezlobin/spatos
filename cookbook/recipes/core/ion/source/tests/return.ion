fn ex
    let a = 1
    if true
        if true
            if true
                while true
                    return $a
                    echo should not be printed
                end
                echo should not be printed
            end
            echo should not be printed
        end
        echo should not be printed
    end
    echo should not be printed
end

fn test_for
    while true
        return
        echo should not be printed
    end
    echo should not be printed
end

fn ex_wrapper
    if ex
        echo should not be printed
    else
        echo should be printed
    end
    if test_for
        echo should be printed
    else
        echo should not be printed
    end
end

ex_wrapper
echo should$(if true; return; else printf ' not'; end) be printed
echo should be printed
