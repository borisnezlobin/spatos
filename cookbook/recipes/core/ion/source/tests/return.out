should be printed
should be printed
should be printed
should be printed
